core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -324,6 +324,21 @@ public String getUrl() {
         }
 
         /**
+         * URL of specific updateSite.
+         */
+        public String getUrl(UpdateSite site) {
```

Comment:
would this not be better defined in UpdateSite so that subclasses can change the location?